### PR TITLE
MES-4660: Making sure that first attempt was recorded when speed not met

### PR DESCRIPTION
**Pull request by szabikr**
## Description

Implementing the correct ent test modal logic

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/test-report/cat-a-mod1/components/speed-check/speed-check.ts [hunk 1]
```diff
@@ -285,23 +285,27 @@ export class SpeedCheckComponent {
     return this.outcome === CompetencyOutcome.D;
   }
 
-  onFirstAttemptChange = (attemptedSpeed: number): void => {
+  onFirstAttemptChange = (attemptedSpeed: string): void => {
+    const firstAttempt = attemptedSpeed === '' ? undefined : Number(attemptedSpeed);
```
Comment (szabikr) [inline src/pages/test-report/cat-a-mod1/components/speed-check/speed-check.ts:289]: @appleabs here is the line which will solve the problem that you are facing.

Comment (appleabs) [reply src/pages/test-report/cat-a-mod1/components/speed-check/speed-check.ts:289]: Ah ok thanks!

**Review by MarkPatt89 — COMMENTED** [review]
Looks good, you do have a `fdescribe` left in though.

File: src/providers/test-report-validator/__tests__/test-report-validator.spec.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ import { configureTestSuite } from 'ng-bullet';
 import { SpeedCheckState } from '../test-report-validator.constants';
 import { CompetencyOutcome } from '../../../shared/models/competency-outcome';
 
-describe('TestReportValidator', () => {
+fdescribe('TestReportValidator', () => {
```
Comment (TomBillingtonUK) [inline src/providers/test-report-validator/__tests__/test-report-validator.spec.ts]: fdescribe

**Review by TomBillingtonUK — APPROVED** [review]
looks good just one comment

File: src/providers/test-report-validator/__tests__/test-report-validator.spec.ts (hunk 2 above)
Comment (szabikr) [reply src/providers/test-report-validator/__tests__/test-report-validator.spec.ts]: darn

**Review by MarkPatt89 — APPROVED** [review]

**Pull request merged**